### DISCUSSION TITLE
Vercel Builds Failing for No Apparent Reason

### DISCUSSION THREAD

**Discussion by TheNickm2** [vercel/vercel#8195] [Help] [answered]
Builds in Vercel for one of my projects started failing without any apparent reason about 10 hours ago. The build logs don't provide any useful information. The entire NextJS build process completes as usual, and then Vercel hangs for about 15-20 minutes before marking the build as "Errored." The build output log has no useful information and simply states that the log is over 4MB so it can't be displayed, yet the entire log is only about 15kb before this message appears:
```
Traced Next.js server files in: 362.722ms
Created all serverless functions in: 1.364s
Collected static files (public/, static/, .next/static): 12.291ms
Build Completed in /vercel/output [5m]
<this is where Vercel appears to be doing nothing for about 15-20 minutes>
Warning: The log length exceeded the limit of 4 MB. Learn more: https://vercel.link/build-log-size-limit
```

The build failure message reads `An unexpected error happened when running this build. We have been notified of the problem. If you have any questions, please contact support@vercel.com.`

Absolutely no error information is being pushed into Logtail. Builds are working fine on my local machine as well as coworker machines and personal servers.

No codebase changes occurred when these builds started failing. The only changes that could have been made at the time of the build failures would be through a headless CMS triggering builds via webhook, however builds are failing for all branches - not just the main branch that is connected to the production headless CMS instance (preview branches use a preview headless instance).

NodeJS version: 14.x
NextJS version: 12.1.0

I'm at a loss with this particular issue. I've checked another of my NextJS sites using a nearly-identical tech stack, and it's building fine. As far as I can tell from the logs, the NextJS build is functioning as expected, and something is erroring out on the Vercel side... but I'm not sure why it's only affecting this one particular site.

Any advice on how to debug this would be appreciated.

**Comment by amyegan** [comment]
Hi @TheNickm2. Thank you for reaching out, and I'm sorry you're running into this error. To help me locate the cause, can you identify any other differences between the sites? For example, 
- Are the other sites using a different CMS? 
- Are all the sites similar in size, or does one have many more pages to build?
- Have any project settings (domains, environment variables, etc) changed recently?
- Do you have any keys for external systems that may have expired?
- Are the successful local builds also taking longer than usual?

Anything you can identify as differences will help me know where to look.

**Reply by TheNickm2** [reply]
Hi there! Thank you for your response. Answers to your questions below:

- Are the other sites using a different CMS? *No, the sites are using the same headless CMS (Storyblok)*
- Are all the sites similar in size, or does one have many more pages to build? *The failing site is quite a bit larger in size than the succeeding site. However, this site has been a rather large size for several months and has built without issues for a long time (only build errors we generally see are when we break something in our codebase, which doesn't appear to be the case in this instance)*
- Have any project settings (domains, environment variables, etc) changed recently? *Last environment variable change was 81 days ago, domains have not changed recently. as far as i'm aware, nothing changed in our config files*
- Do you have any keys for external systems that may have expired? *All API keys [shopify, storyblok, algolia] seem to be valid as far as I can tell*
- Are the successful local builds also taking longer than usual? *Local builds haven't changed much in terms of build duration*

Additional info regarding the size of the site - it is generating a fairly large number of pages, our typical Vercel builds take about 4 minutes to build & deploy. With the current issue that's occurring, the NextJS build is still taking about 4-5 minutes based on the Vercel logs, but something appears to be happening after the NextJS build step. I can verify based on the logging output that my `build` and `postbuild` scripts are both fully executing before Vercel begins to hang.

Please feel free to ask if you need any additional information. Thank you again for taking a look & responding! :)

**Reply by amyegan** [reply]
Thanks for sharing this info! I'll take a look and update here when I know more.

**Comment by swarnava** [comment, marked as answer]
Hi Nick, Can you redeploy once more and let us know if you are facing an error?

**Reply by TheNickm2** [reply]
Hello there! Builds are now completing successfully. Thank you for the assistance!

**Comment by Syesy** [comment]
Hi @swarnava we have the same issue
We're using 
NodeJS version: 14.x
NextJS version: 12.0.9
Can you give us some advice about that? thank you

**Reply by swarnava** [reply]
Hi, can you create a separate discussion with deployment link? Thanks!
